ulib/axlibc/Cargo.toml

```diff
@@ -23,47 +23,73 @@ crate-type = ["staticlib"]
 default = []
 
 # Multicore
-smp = ["axstd/smp"]
+smp = ["axfeat/smp", "arceos_posix_api/smp"]
 
 # Floating point/SIMD
-fp_simd = ["axstd/fp_simd"]
+fp_simd = ["axfeat/fp_simd", "arceos_posix_api/fp_simd"]
 
 # Memory
-alloc = ["axstd/alloc", "dep:axalloc"]
+alloc = ["axfeat/alloc", "dep:axalloc", "arceos_posix_api/alloc", "axio/alloc"]
+alloc-tlsf = ["axfeat/alloc-tlsf"]
+alloc-slab = ["axfeat/alloc-slab"]
+alloc-buddy = ["axfeat/alloc-buddy"]
+paging = ["axfeat/paging"]
+tls = ["axfeat/tls"]
 
 # Multi-task
-multitask = ["axstd/multitask", "axtask/multitask"]
+multitask = ["axfeat/multitask", "axtask/multitask", "arceos_posix_api/multitask"]
+sched_fifo = ["axfeat/sched_fifo"]
+sched_rr = ["axfeat/sched_rr"]
+sched_cfs = ["axfeat/sched_cfs"]
 
 # File system
-fs = ["axstd/fs", "fd"]
+fs = ["axfeat/fs", "fd", "arceos_posix_api/fs"]
+myfs = ["axfeat/myfs"]
 
 # Networking
-net = ["axstd/net", "dep:axnet", "fd"]
+net = ["axfeat/net", "dep:axnet", "fd", "arceos_posix_api/net"]
+dns = []
+
+# Display
+display = ["axfeat/display"]
+
+# Device drivers
+bus-mmio = ["axfeat/bus-mmio"]
+bus-pci = ["axfeat/bus-pci"]
+driver-ramdisk = ["axfeat/driver-ramdisk"]
+driver-ixgbe = ["axfeat/driver-ixgbe"]
+driver-bcm2835-sdhci = ["axfeat/driver-bcm2835-sdhci"]
+
+# Logging
+log-level-off = ["axfeat/log-level-off"]
+log-level-error = ["axfeat/log-level-error"]
+log-level-warn = ["axfeat/log-level-warn"]
+log-level-info = ["axfeat/log-level-info"]
+log-level-debug = ["axfeat/log-level-debug"]
+log-level-trace = ["axfeat/log-level-trace"]
 
 # Libc features
 fd = ["alloc"]
-pipe = ["fd"]
-select = ["fd"]
-epoll = ["fd"]
+pipe = ["fd", "arceos_posix_api/pipe"]
+select = ["fd", "arceos_posix_api/select"]
+epoll = ["fd", "arceos_posix_api/epoll"]
 
 [dependencies]
 # ArceOS modules
 axfeat = { path = "../../api/axfeat" }
-axstd = { path = "../axstd" }
+arceos_posix_api = { path = "../../api/arceos_posix_api" }
+axerrno = { path = "../../crates/axerrno" }
+axio = { path = "../../crates/axio" }
+flatten_objects = { path = "../../crates/flatten_objects" }
+axconfig = { path = "../../modules/axconfig" }
```

Comment:
Try to remove module dependencies in axlibc (axconfig, axhal, axnet, ...)

Reply:
it seems these modules are not actually used expect axconfig, we can add `mod config` in `arceos_posix_api` that export constants in axconfig.

Reply:
Yes, only `axconfig` is used in build.rs to generate `axconfig.h`, and is used in ax_sysconf.
I move `axconfig.h` generating to `arceos_posix_api/build.rs`.

Reply:
Another question: where to put `axconfig.h` and `ax_pthread_mutex.h`? Now that these two files are generated by `arceos_posix_api/build.rs`, is it better to put this two files in `arceos-posix-api/`?

Reply:
`axconfig.h` can be removed, remain `ax_pthread_mutex.h` unchanged and let me think.

Reply:
> it seems these modules are not actually used expect axconfig, we can add `mod config` in `arceos_posix_api` that export constants in axconfig.

In `sysconf()`, this function not only access `axconfig`, but also `axalloc::global_allocator()` to get available number of pages. So I make another syscall: `sys_sysconf` to get this infomation.